pkgs/os-specific/linux/decklink/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv
+, lib
+, blackmagicDesktopVideo
+, kernel
+}:
+
+stdenv.mkDerivation rec {
+  pname = "decklink";
+
+  # the download is a horrible curl mess. we reuse it between the kernel module
+  # and desktop service, since the version of the two have to match anyways.
+  # See pkgs/tools/video/blackmagic-desktop-video/default.nix for more.
+  inherit (blackmagicDesktopVideo) src version;
+
+  KERNELDIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
+  INSTALL_MOD_PATH = placeholder "out";
+
+  nativeBuildInputs =  kernel.moduleBuildDependencies;
+
+  setSourceRoot = ''
+    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}/other/x86_64/desktopvideo-${version}-x86_64.tar.gz
+    sourceRoot=$NIX_BUILD_TOP/desktopvideo-${version}-x86_64/usr/src
```

Comment:
```suggestion
    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}/other/${stdenv.hostPlatform.uname.processor}/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}.tar.gz
    sourceRoot=$NIX_BUILD_TOP/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}/usr/src
```

Reply:
Actually, we since we cannot template the download URL, this would be a major pain to do. While Blackmagic has started to offer ARM64 builds, they are exposed under a different download.

Reply:
Just checked, misremembered. Applied the suggestion.